themes/src/main/resources/theme/keycloak.v2/account/src/app/content/linked-accounts-page/LinkedAccountsPage.tsx

```diff
@@ -200,33 +210,34 @@ class LinkedAccountsPage extends React.Component<LinkedAccountsPageProps, Linked
         )
     }
 
-    private badge(account: LinkedAccount): React.ReactNode {
+    private label(account: LinkedAccount): React.ReactNode {
         if (account.social) {
-            return (<Badge><Msg msgKey='socialLogin'/></Badge>);
+            return (<Label color="blue"><Msg msgKey='socialLogin'/></Label>);
         }
 
-        return (<Badge style={{backgroundColor: "green"}} ><Msg msgKey='systemDefined'/></Badge>);
+        return (<Label color="green"><Msg msgKey='systemDefined'/></Label>);
     }
 
     private findIcon(account: LinkedAccount): React.ReactNode {
-        const socialIconId = `${account.providerAlias}-idp-icon-social`;
-        if (account.providerName.toLowerCase().includes('github')) return (<GithubIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('linkedin')) return (<LinkedinIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('facebook')) return (<FacebookIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('google')) return (<GoogleIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('instagram')) return (<InstagramIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('microsoft')) return (<MicrosoftIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('bitbucket')) return (<BitbucketIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('twitter')) return (<TwitterIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('openshift')) return (<OpenshiftIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('gitlab')) return (<GitlabIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('paypal')) return (<PaypalIcon id={socialIconId} size='xl'/>);
-        if (account.providerName.toLowerCase().includes('stackoverflow')) return (<StackOverflowIcon id={socialIconId} size='xl'/>);
-
-        return (<CubeIcon id={`${account.providerAlias}-idp-icon-default`} size='xl'/>);
+      const socialIconId = `${account.providerAlias}-idp-icon-social`;
+      console.log(account);
+      switch (true) {
```

Comment:
This code cannot be right, none of the cases besides the `default` will ever match.